chef/automate
# Infra proxy API to get the organisation of chef server

### :nut_and_bolt: Description: What code changed, and why?
List the organisation from the chef server
To use the web ui key in request to the chef server, I have added one parameter ```isWebuiKey```.
Based on this parameter we are adding one header as ```X-Ops-Request-Source:web``` in the go-chef library while making request to the chef server.
For go-chef changes, please refer this PR https://github.com/chef/go-chef/pull/30



### :chains: Related Resources
https://github.com/chef/automate/issues/6161
https://github.com/chef/go-chef/issues/29

### :+1: Definition of Done
I have added an API to list the organisation from the chef server
and 
### :athletic_shoe: How to Build and Test the Change
Steps to build:
```
rebuild components/infra-proxy-service
rebuild components/automate-gateway
```
Steps to test:

1. Ensure Chef Infra Server is running. If it is not start it with ```start_chef_server```
2. Add the chef-server test environment credentials and extract the SERVER IDs
3. Add the web ui key in the request body

Request
This is a post request.
```curl -sSkH "api-token: $(get_admin_token)" 'https://a2-dev.test/api/v0/infra/servers/SERVER_ID/automateinfraserverorgs'```

request body
```
{
    "webui_key":"WEBUI_KEY"
}
```

Response

```
{
    "orgs": [
        {
            "id": "test",
            "name": "test",
            "admin_user": "",
            "credential_id": "",
            "server_id": "local-dev",
            "projects": []
        }
    ]
}
```

### :white_check_mark: Checklist

**All PRs** must tick these:


With occasional exceptions, all PRs **from Progress employees** must tick these:


All PRs **from Progress employees** should tick these if appropriate:


*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
